### PR TITLE
Use logging stream image to enable ES6 for EO and CLO

### PR DESCRIPTION
**Pull request by periklis**

### CODE REVIEW

File: hack/testing/utils [hunk 1]
```diff
@@ -221,20 +221,27 @@ function deploy_clusterlogging_operator() {
     local manifest=${repo_dir}/manifests
     local version=$(basename $(find $manifest -type d | sort -r | head -n 1))
     local csv=$manifest/$version/cluster-logging.*.clusterserviceversion.yaml
-    local es_img k_img c_img f_img op_img
+    local es_img k_img c_img f_img op_img    
+    local es_version="5"
+    
+    LOGGING_IMAGE_STREAM=${LOGGING_IMAGE_STREAM:-$version}
+    if [ "$LOGGING_IMAGE_STREAM" != "$version" ]; then
+        es_version="6"
```
Comment (jcantrill) [inline hack/testing/utils]: This is not needed.  This branch will always utilize ES6.  All components should be pegged to by kibana/elasticsearch6 with the exception of curator.  There is no "6" version of it

File: hack/testing/utils [hunk 2]
```diff
@@ -221,20 +221,27 @@ function deploy_clusterlogging_operator() {
...
+    fi
+        
     if [ -n "${IMAGE_FORMAT:-}" ] ; then
         IMAGE_CLUSTER_LOGGING_OPERATOR=$(sed -e "s,\${component},cluster-logging-operator," <(echo $IMAGE_FORMAT))
-        es_img=${IMAGE_FORMAT/'${component}'/logging-elasticsearch5}
-        k_img=${IMAGE_FORMAT/'${component}'/logging-kibana5}
-        c_img=${IMAGE_FORMAT/'${component}'/logging-curator5}
+        es_img=${IMAGE_FORMAT/'${component}'/logging-elasticsearch$es_version}
+        k_img=${IMAGE_FORMAT/'${component}'/logging-kibana$es_version}
+        c_img=${IMAGE_FORMAT/'${component}'/logging-curator$es_version}
```
Comment (jcantrill) [inline hack/testing/utils]: can only be 'curator5'

File: hack/testing/utils [hunk 3]
```diff
@@ -221,20 +221,27 @@ function deploy_clusterlogging_operator() {
...
+    if [ "$LOGGING_IMAGE_STREAM" != "$version" ]; then
+        es_version="6"
+    fi
+        
     if [ -n "${IMAGE_FORMAT:-}" ] ; then
         IMAGE_CLUSTER_LOGGING_OPERATOR=$(sed -e "s,\${component},cluster-logging-operator," <(echo $IMAGE_FORMAT))
-        es_img=${IMAGE_FORMAT/'${component}'/logging-elasticsearch5}
-        k_img=${IMAGE_FORMAT/'${component}'/logging-kibana5}
-        c_img=${IMAGE_FORMAT/'${component}'/logging-curator5}
+        es_img=${IMAGE_FORMAT/'${component}'/logging-elasticsearch$es_version}
```
Comment (jcantrill) [inline hack/testing/utils]: peg to 6

File: vendor/github.com/openshift/elasticsearch-operator/Makefile [hunk 4]
```diff
@@ -18,7 +18,10 @@ KUBECONFIG?=$(HOME)/.kube/config
 MAIN_PKG=cmd/manager/main.go
 RUN_LOG?=elasticsearch-operator.log
 RUN_PID?=elasticsearch-operator.pid
-
+LOGGING_IMAGE_STREAM?=feature-es6x
```
Comment (jcantrill) [inline vendor/github.com/openshift/elasticsearch-operator/Makefile]: Does this PR also need a EO update?

File: hack/testing/utils (hunk 1 above)
Comment (periklis) [reply hack/testing/utils]: Good catch!

File: hack/testing/utils (hunk 3 above)
Comment (periklis) [reply hack/testing/utils]: Done

File: hack/testing/utils (hunk 2 above)
Comment (periklis) [reply hack/testing/utils]: Done

File: vendor/github.com/openshift/elasticsearch-operator/Makefile (hunk 4 above)
Comment (periklis) [reply vendor/github.com/openshift/elasticsearch-operator/Makefile]: No need to EO update

**Comment by periklis** [comment]
@jcantrill Regarding the failing e2e test, I could track the cause down to a wrong ImageStream reference for ES6 images (e.g. `stable:logging-elasticsearch6` should be `feature-es6x:logging-elasticsearch6`). However, I must admit I don't have a clue how to extend the jobs config in the `release/ci-operator` repo as mentioned during your last architecture lightning talk. AFAICS, in the build [imagestreams.json](https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-logging-operator/368/pull-ci-openshift-cluster-logging-operator-feature-es6x-e2e-operator/22/artifacts/build-resources/imagestreams.json) we miss everything that gets promoted from [OAL feature-es6x config](https://github.com/openshift/release/blob/master/ci-operator/config/openshift/origin-aggregated-logging/openshift-origin-aggregated-logging-feature-es6x.yaml#L58). Any idea how to proceed here?

**Comment by jcantrill** [comment]
@periklis closing this for now.  we need to add the one fix into master regarding the alias'.  The other changes we should consider making as general CI improvements

**Pull request closed without merging**